### PR TITLE
Remove old applyStashedOp model, and only support the new model.

### PR DESCRIPTION
**Pull request by anthony-murphy**
## Background
This change is related to https://github.com/microsoft/FluidFramework/pull/19518. That change introduced a new and simpler mechanism for applying stash ops. Specifically, it allows dds to reuse their existing flows for creating local changes, rather than needing custom flows to apply the change and get the local op metadata. This reduces complexity, bundle size, and bug potential by having fewer code paths to do the same thing; apply changes locally.

## Overview
Now that all dds in our repo have been updated to the new stashing model, this change remove support for the old model, cleans up some apis, enables some tests, and enables stashing in stress by default.

### CODE REVIEW

File: packages/dds/map/src/test/mocha/directoryFuzzTests.spec.ts [hunk 1]
```diff
@@ -394,7 +396,7 @@ describe("SharedDirectory fuzz", () => {
 			// was refactored to use the DDS fuzz harness.
 			maxNumberOfClients: Number.MAX_SAFE_INTEGER,
 			clientAddProbability: 0.08,
-			stashableClientProbability: 0.2,
+			stashableClientProbability: undefined,
```
Comment (anthony-murphy) [inline packages/dds/map/src/test/mocha/directoryFuzzTests.spec.ts:399]:
```suggestion
			stashableClientProbability: 0.2,
```

File: packages/dds/shared-object-base/src/sharedObject.ts [hunk 2]
```diff
@@ -537,21 +537,23 @@ export abstract class SharedObjectCore<TEvent extends ISharedObjectEvents = ISha
...
-	 * with pending changes. This prepares the SharedObject for seeing an ACK
-	 * for the op or resubmitting the op upon reconnection.
-	 * @param content - Contents of a stashed op.
-	 * @returns Should return void.
+	 * with pending changes. The rehydration process replays all remote ops
+	 * and applies stashed ops after the remote op with a sequence number
+	 * that matches that of the stashed op is applied. This ensures
+	 * stashed ops are applied at the same state they were originally created.
+	 *
+	 * It is possible that stashed ops have been sent in the past, and will be found when
```
Comment (dannimad) [inline packages/dds/shared-object-base/src/sharedObject.ts:547]: Sometimes when I read documentation about stashed ops, it's easy to get confused with many terms (stashed - pending / remote - saved). Also, I would love an independent md file to understand more about the internal flow of applyStashedOp and not have to look at PR descriptions.

**Review by dannimad — APPROVED** [review]
I don't think current build failures are related to the PR so lgtm.

File: packages/dds/shared-object-base/src/sharedObject.ts (hunk 2 above)
Comment (dannimad) [reply packages/dds/shared-object-base/src/sharedObject.ts:547]: This md file could have a glossary about all terms being used.

**Pull request merged**